src/khepri_pattern_tree.erl

```diff
@@ -255,6 +257,36 @@ fold_data(PatternTreeNode, CurrentPath, FoldFun, Acc) ->
             FoldFun(CurrentPath, Payload, Acc)
     end.
 
+-spec fold(PatternTree, Fun, Acc) -> Ret when
+      PatternTree :: khepri_pattern_tree:tree(Payload),
+      Fun :: fold_fun(Payload),
+      Acc :: fold_acc(),
+      Ret :: fold_acc(),
+      Payload :: payload().
+%% @doc Folds over the pattern tree passing each pattern, payload and the
+%% accumulator to the fold function.
+%%
+%% Unlike {@link fold_matching/5} all payloads are passed to the fold function.
+
+fold(#pattern_node{} = PatternTree, Fun, Acc) ->
+    fold(PatternTree, Fun, [], Acc).
+
+fold(
+  #pattern_node{payload = Payload, child_nodes = ChildNodes},
+  Fun, ReversedPath, Acc0) ->
```

Comment:
Style comment: could you please name the argument `Acc` and then increment the suffix in the function body? I like that the arguments have the "normal" name, the same as the type spec and the doc. I know, this is the opposite of Ra where the normal name is used for the returned value :-)